httpx/_auth.py

```diff
@@ -51,10 +54,10 @@ class FunctionAuth(Auth):
     """
 
     def __init__(self, func: typing.Callable[[Request], Request]) -> None:
-        self.func = func
+        self._func = func
```

Comment:
Curious if we'd want to more clearly separate out "the Auth interface" from our default "function-based Auth" implementation - ie separate Auth from FuncAuth?

Reply:
I'm not sure *exactly* what you mean here, but... We don't expose `FunctionAuth` publicly anywhere, but we *might* want to have a property setter on Client, as we do with several of the other attributes on it, so that eg...

```
The following are all valid...

client.auth = None
client.auth = ('username', 'password')
client.auth = httpx.DigestAuthentiction(...)
client.auth = simple_function_auth
client.auth = CustomAuthClass()

print(client.auth)  # Always returns an instance of `Auth`. Our instances don't expose any additional public API beyond that.
```

Reply:
Ah okay, nevermind — the GitHub UI tricked me. I thought this diff was on the `Auth` class, but it's on `FunctionAuth`. So it's all as I was meaning to think it should be. :-)